### PR TITLE
helm: Do not render empty fields (subPath, args, env & envFrom)

### PR DESCRIPTION
**Pull request by adberger**
#### What this PR does

Many fields in the mimir-distributed helm chart are templated, even if no values are specified.
This PR should fix these.

#### Checklist

- [ ] Tests updated.
- [ ] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [ ] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

**Comment by adberger** [comment]
@dimitarvdimitrov I'm not sure how to proceed from here.  
I've ran `make build-helm-tests` and pushed a new commit, but since upstream was already updated there is a conflict.
What's the best approach?

**Comment by dimitarvdimitrov** [comment]
we don't have strict git instructions. But what I'd do is 

1. squash your 4 commits into 1 (`git rebase -i HEAD~4` and then mark all of them as `f` or `s`)
2. make sure your local `upstream:main` ref is up to date 
2. rebase onto `upstream:main`
3. force-push to `adberger:main`

**Comment by adberger** [comment]
@dimitarvdimitrov I tried my best, can you rerun the tests?

File: operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier/graphite-querier-dep.yaml [hunk 1]
```diff
@@ -84,26 +86,33 @@ spec:
...
               {{- toYaml . | nindent 12 }}
             {{- end }}
             {{- with .Values.admin_api.extraEnvFrom }}
               {{- toYaml . | nindent 12 }}
             {{- end }}
+<<<<<<< HEAD
       {{- with .Values.graphite.querier.nodeSelector }}
+=======
+          {{- end }}
+>>>>>>> ce221d3f2 (fix: Only template persistence subPath if it exists for all components)
```
Comment (dimitarvdimitrov) [inline operations/helm/charts/mimir-distributed/templates/graphite-proxy/graphite-querier/graphite-querier-dep.yaml:115]: this looks like a bad merge. I can spot a few similar ones

**Comment by adberger** [comment]
Apologies, I gave it another try.

**Review by dimitarvdimitrov — APPROVED** [review]
LGTM, thank you. I pushed a commit with a changelog entry

**Pull request merged**